nautobot_chatops/models.py

```diff
@@ -9,6 +9,26 @@
 
 from .choices import AccessGrantTypeChoices, CommandStatusChoices, CommandTokenPlatformChoices
 
+import re
+
+
+class QuerySetRegexMatch(models.QuerySet):
+    """Customized QuerySet to match string agains regexes stored in database.
+
+    Return a list of matched records.
+    """
+
+    def match_re(self, value):
+        records = self.all()
+
+        results = []
+        for row in records:
+            if row.subcommand == "*":  # skip as it is not a valid regex expression and would raise exception
+                continue
+            if re.match(row.subcommand, value):
+                results.append(row.subcommand)
+        return results
```

Comment:
I'm definitely a bit concerned about potential performance impact of this design. It *should* be possible to do this as an actual DB query, rather than querying for `all()` and then iterating through the results, though from some quick Googling it looks like custom SQL might be needed to make that possible...